src/components/forms/TextInput/TextInput.test.tsx

```diff
@@ -11,6 +11,47 @@ describe('TextInput component', () => {
     expect(queryByTestId('textInput')).toBeInTheDocument()
   })
 
+  describe('masking features', () => {
+    const setup = () => {
+      const utils = render(
+        <TextInput
+          id="input-type-alphanumeric"
+          name="input-type-alphanumeric"
+          type="text"
+          aria-describedby="hint-alphanumeric"
+          mask="___ ___"
+          pattern="\w\d\w \d\w\d"
+          charset="A#A #A#"
+        />
+      )
+      const input = screen.getByTestId('textInput')
+      const mask = screen.getByTestId('input-type-alphanumericMask')
+      return {
+        input,
+        mask,
+        ...utils,
+      }
+    }
+
+    it('renders with class and placeholder', () => {
+      const { input, mask } = setup()
+      expect(input).toHaveClass('usa-masked')
+      expect(mask).toHaveTextContent('___ ___')
+    })
+
+    it('autoformats inputted text', () => {
+      const { input } = setup()
+      fireEvent.change(input, { target: { value: 'A1B2C3' } })
+      expect((input as HTMLInputElement).value).toBe('A1B 2C3')
+    })
+
+    it('rejects entry based on charset', () => {
+      const { input } = setup()
+      fireEvent.change(input, { target: { value: 'A1B 2CC' } })
+      expect((input as HTMLInputElement).value).toBe('A1B 2C')
+    })
+  })
```

Comment:
**praise**: ❤️ ❤️ ❤️ to this style of testing. So glad it's catching on. 

([If anyone else is curious about it](https://derekndavis.com/posts/maintainable-testing-react-testing-library))